build.gradle

```diff
@@ -47,6 +47,23 @@ jacoco {
     toolVersion = "0.7.5.201505241946"
 }
 
+final requiredJavaVersion = "8"
+final buildPrerequisitesMessage = "See https://github.com/broadinstitute/picard/blob/master/README.md#building-picard for information on how to build picard"
+// Ensure that we have the right JDK version, a clone of the git repository
+def ensureBuildPrerequisites(requiredJavaVersion, buildPrerequisitesMessage) {
+    // Make sure we can get a ToolProvider class loader. If not we may have just a JRE, or a JDK from the future.
+    if (ToolProvider.getSystemToolClassLoader() == null) {
```

Comment:
I'm not super clear on the differences between JDK versions... Is `ToolProvider` something new in 8, or could this check pass for 7? If it'll pass, is there a way we can detect 8?

Reply:
@cmnbroad - you did this change in GATK. Do you know something about it?

Reply:
This was put in to protect against NPEs in the case where someone tries to build using a JRE (it has happened) or tries to use a JDK from the future (I haven't tried it but I think it will fail on JDK9). The build file has a `targetCompatibility` declaration that should prevent building with older JDKs.